apps/dashboard/features/governance/hooks/useProposals.ts

```diff
@@ -0,0 +1,214 @@
+import { useCallback, useMemo, useState, useEffect } from "react";
+import { ApolloError } from "@apollo/client";
+import { DaoIdEnum } from "@/shared/types/daos";
+import {
+  useGetProposalsQuery,
+  GetProposalsQuery,
+  QueryInput_Proposals_OrderDirection,
+  QueryProposalsArgs,
+} from "@anticapture/graphql-client/hooks";
+import type { Proposal as GovernanceProposal } from "@/features/governance/types";
+import { transformToGovernanceProposal } from "@/features/governance/utils/transformToGovernanceProposal";
+
+export interface PaginationInfo {
+  hasNextPage: boolean;
+  totalCount: number;
+  currentPage: number;
+  totalPages: number;
+  itemsPerPage: number;
+  currentItemsCount: number;
+}
+
+export interface UseProposalsResult {
+  proposals: GovernanceProposal[];
+  loading: boolean;
+  error: ApolloError | undefined;
+  pagination: PaginationInfo;
+  fetchNextPage: () => Promise<void>;
+  isPaginationLoading: boolean;
+}
+
+// Use the generated GraphQL arguments type and extend with pagination
+export interface UseProposalsParams
+  extends Omit<QueryProposalsArgs, "skip" | "limit"> {
+  itemsPerPage?: number;
+}
+
+export const useProposals = ({
+  fromDate,
+  orderDirection = QueryInput_Proposals_OrderDirection.Desc,
+  status,
+  itemsPerPage = 10,
+}: UseProposalsParams = {}): UseProposalsResult => {
+  const [isPaginationLoading, setIsPaginationLoading] = useState(false);
+  const [allProposals, setAllProposals] = useState<GovernanceProposal[]>([]);
+
+  const queryVariables = useMemo(
+    () => ({
+      skip: 0, // Always start from 0 for first query
+      limit: itemsPerPage,
+      orderDirection,
+      status,
+      fromDate,
+    }),
+    [itemsPerPage, orderDirection, status, fromDate],
+  );
+
+  // Main proposals query
+  const { data, loading, error, fetchMore } = useGetProposalsQuery({
+    variables: queryVariables,
+    notifyOnNetworkStatusChange: true,
+    context: {
+      headers: {
+        "anticapture-dao-id": DaoIdEnum.ENS,
```

Comment:
even though it is gonna be only exposed for ENS, please make this a generic code so we have fewer things to change when we decide to support it for the next DAO